src/login/logind-button.c

```diff
@@ -344,10 +345,12 @@ int button_open(Button *b) {
         r = button_suitable(fd);
         if (r < 0)
                 return log_warning_errno(r, "Failed to determine whether input device %s is relevant to us: %m", p);
-        if (r == 0)
+        if (r == 0) {
+                b->fd = TAKE_FD(fd);
```

Comment:
What's the point of this? Can you explain? Why would we keep an fd open to this, if we cannot use it?

Reply:
If we don't keep the fd open here, the device is freed at the end of the function before returning to manager_process_button_device. It takes 30ms on my computer to do that.
When the device isn't relevant, manager_process_button_device call button_free right after the return of button_open. 
I though it was cleaner to free the device in button_free than to make a call to asynchronous_close() in button_open.

Reply:
It wouldn't be cleaner to replace `_cleanup_close_` by a `cleanup` function calling `asynchronous_close`?
This way all exit paths will close asynchronously the device.

As I understand the convention of systemd code, when a function is returning an error (negative value), the function should not have an external side effect. Currently this code is fine, but may lead to future bug.